src/mount/fuse/main.cc

```diff
@@ -427,6 +427,59 @@ static size_t count_colons_in_str(const char *str, size_t len) {
 	return colons_count;
 }
 
+/**
```

Comment:
Consider this Copilot suggestions as possible improvements for your code. This version uses modern `std::string` and STL iterators to make easier and more readable the code.

Please, feel free to test and make the changes as you consider more correct.
```
#include <cctype> // for std::tolower
#include <cstring> // for strlen, strcmp
#include <string> // for std::string

// Normalize a single argument's casing
void normalize_argument_casing(char *arg) {
    std::string argument(arg);
    bool should_apply_case_change = true;

    for (auto &c : argument) {
        if (c == ',') {
            should_apply_case_change = true;
        } else if (c == '=') {
            should_apply_case_change = false;
        } else if (should_apply_case_change) {
            c = std::tolower(c);
        }
    }

    // Copy the modified string back to the original C-string argument
    std::copy(argument.begin(), argument.end(), arg);
    arg[argument.size()] = '\0'; // Null-terminate the modified C-string
}

// Normalize options casing
void normalize_options_casing(fuse_args &args) {
    bool is_previous_dash_o = false;

    for (int index = 1; index < args.argc; ++index) {
        std::string current_arg(args.argv[index]);
        if (is_previous_dash_o || (current_arg.length() > 2 && current_arg[0] == '-' &&
                                   (current_arg[1] == 'o' || current_arg[1] == '-'))) {
            if (!is_previous_dash_o) {
                // Skip the initial "-o" or "--" prefix
                current_arg = current_arg.substr(2);
                normalize_argument_casing(&args.argv[index][2]);
            } else {
                normalize_argument_casing(args.argv[index]);
            }
            is_previous_dash_o = false;
        } else {
            is_previous_dash_o = (current_arg == "-o");
        }
    }
}
```

Reply:
Done, now is more readable thanks to this approach.